pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -53,6 +53,8 @@
 , coreSetup ? false # Use only core packages to build Setup.hs.
 , useCpphs ? false
 , hardeningDisable ? lib.optional (ghc.isHaLVM or false) "all"
+, hasDataDir ? true
```

Comment:
10933 out of 11949 Haskell packages have `hasDataDir = false` since they don't install any data files. Therefore, `false` should be the default -- not `true` -- so that we omit this attribute in 91% of all Haskell Nix expressions.

0d862e9f09a5fc4bf8414fa405df22f0d7c6984e adds that attribute to the package set for *all* expressions -- regardless of whether it's true or false -- so that we can switch the default in the generic builder easily.